app/controller/button/SpatialQueryButton.js

```diff
@@ -84,7 +101,9 @@ Ext.define('CpsiMapview.controller.button.SpatialQueryButtonController', {
         if (!view.queryLayer) {
             return;
         }
-        var mapComp = CpsiMapview.view.main.Map.guess();
+
+        var mapComp = me.mapComponent || BasiGX.util.Map.getMapComponent();
```

Comment:
```suggestion
        if (!me.mapComponent) {
            me.mapComponent = BasiGX.util.Map.getMapComponent();
        }
```

Reply:
As `mapComponent` is not set currently, I would prefer to to set `me.mapComponent` where and use `me.mapComponent` afterwards, what to you think?

Reply:
Yes that makes sense.
The [DigitizeButton](https://github.com/compassinformatics/cpsi-mapview/blob/master/app/controller/button/DigitizeButtonController.js#L289) currently uses:

`var mapComponent = me.mapComponent || BasiGX.util.Map.getMapComponent();`

It would make sense to harmonize the approach for all tools (also with regards to `CpsiMapview.view.main.Map.guess()`)

Reply:
:+1: